src/aleph/http/client.clj

```diff
@@ -487,7 +712,241 @@
                nil
                nil))))
 
+
+
+(defn- rsp-handler
+  "Returns a fn that takes a response map and returns the final Ring response map.
+
+   Shared between HTTP versions.
+   Handles errors, closing, and converts the body if not raw."
+  [{:keys [ch keep-alive? raw-stream? req response-buffer-size t0]}]
+  (fn handle-response [rsp]
+    (cond
+      (instance? Throwable rsp)
+      (d/error-deferred rsp)
+
+      (identical? ::closed rsp)
+      (d/error-deferred
+        (ex-info
+          (format "connection was closed after %.3f seconds" (/ (- (System/nanoTime) t0) 1e9))
+          {:request req}))
+
+      raw-stream?
+      rsp
+
+      :else
+      (d/chain' rsp
+                ;; chain, since conversion to InputStream may take time
+                (fn convert-body [rsp]
+                  (let [body (:body rsp)]
+                    ;; handle connection life-cycle
+                    (when (false? keep-alive?)              ; assume true if nil
+                      (if (s/stream? body)
+                        (s/on-closed body #(netty/close ch))
+                        (netty/close ch)))
+
+                    ;; Since it's not raw, convert the body to an InputStream
+                    (assoc rsp
+                           :body
+                           (bs/to-input-stream body
+                                               {:buffer-size response-buffer-size}))))))))
+
+(defn- http1-req-handler
+  "Returns a fn that takes a Ring request and returns a deferred containing a
+   Ring response.
+
+   Otherwise, puts/takes to/from the requests/responses streams."
+  [{:keys [ch keep-alive? raw-stream? requests response-buffer-size responses]}]
```

Comment:
I would remove all the parameters that are no in use.

Reply:
Yeah, good point. I was shuffling a lot of stuff around, and waiting for the dust to settle, and forgot to clean that up

Reply:
Done